src/merlin.ml

```diff
@@ -31,6 +31,38 @@ module Preprocess = struct
       | _  -> Other
 end
 
+module Dot_file = struct
+  let b = Buffer.create 256
+
+  let printf = Printf.bprintf b
+  let print = Buffer.add_string b
+
+  let to_string ~obj_dirs ~src_dirs ~flags ~ppx ~remaindir =
+    let serialize_path = Path.reach ~from:remaindir in
+    Buffer.clear b;
+    Path.Set.iter obj_dirs ~f:(fun p ->
+      printf "B %s\n" (serialize_path p)
+    );
+    Path.Set.iter src_dirs ~f:(fun p ->
+      printf "S %s\n" (serialize_path p)
+    );
+    begin match ppx with
+    | [] -> ()
+    | ppx ->
+      printf "FLG -ppx %s\n" (Filename.quote (String.concat ~sep:" " ppx));
```

Comment:
Before we were calling `quote_for_shell` for individual element and then concatenating the result, this is a bit different. I think we should be conservative here